docs/pages/versions.js

```diff
@@ -8,3 +10,43 @@ const reqPrefix = 'pages/versions';
 export default function Page() {
   return <MarkdownDocs req={req} reqSource={reqSource} reqPrefix={reqPrefix} />;
 }
+
+async function getBranches() {
+  let branches = [];
+  try {
+    const result = await fetch('https://api.github.com/repos/mui-org/material-ui-docs/branches');
+    branches = await result.json();
+  } catch (err) {
+    // Swallow the exceptions.
```

Comment:
I suspect that running a query to the GitHub API, at each build will quickly get us to rate limit. At the same time, because we build only once, an error here is likely a concern we should pay attention to. I would propose that 1. we don't swallow the error. 2. we don't query for pull request (`process.env.PULL_REQUEST !== 'true'`)  or we use an API token.

Reply:
> For API requests using Basic Authentication or OAuth, you can make up to 5000 requests per hour.

Since this is our only usage of the API so far I doubt we hit the limit on regular PRs. 

I wouldn't swallow it regardless.

Reply:
Fair. +1 for trying without the shallow and see how that goes. My concern is with the shared public IP Netlify might use for its workers. But I'm only guessing 🤷‍♂️.

Reply:
Oh I was under the impression we're using a separate API key. No harm having it public if the key can only execute read-only queries.

Reply:
> I suspect that running a query to the GitHub API, at each build will quickly get us to rate limit. At the same time, because we build only once, an error here is likely a concern we should pay attention to. I would propose that 1. we don't swallow the error. 2. we don't query for pull request (`process.env.PULL_REQUEST !== 'true'`) or we use an API token.


`async function getBranches() {
  let branches = [];
  try {
    if (process.env.PULL_REQUEST !== 'true') {
      const result = await fetch('https://api.github.com/repos/mui-org/material-ui-docs/branches');
      branches = await result.json();
    }
  } catch (err) {
    // eslint-disable-next-line no-console
    console.log(err);
  }

  branches = branches || [];
  return branches;
}`

@oliviertassinari is this your proposal?

Reply:
No, I believe the conclusion of the discussion with @eps1lon is to remove error shallowing and to make an authenticated request, to increase the rate-limiting from 60/hour/IP to 5,000/hour/user.
https://developer.github.com/v3/#rate-limiting. I have added this env variable: GITHUB_AUTH.

Maybe something like this:

```js
fetch('https://api.github.com/repos/mui-org/material-ui-docs/branches', {
	method: 'GET',
	headers: {
		'Authorization': new Buffer(process.env.GITHUB_AUTH).toString('base64'),
	},
})
```

Reply:
Got it.

Reply:
What scope does the token have? I might be misunderstanding, but I can't see one that isn't overly broad: https://developer.github.com/apps/building-oauth-apps/understanding-scopes-for-oauth-apps/

Reply:
Personnal token, read only on repo.

Reply:
It's a personnal token with the `public_repo` scope.

Reply:
It sounds like it's not write. Deleted 🤔

Reply:
So, do we give up on the changes?

Reply:
The issue most likely isn't caused by the versions page itself, but by the GitHub stars counter on the home page. I don't think it matters much if that isn't 100% accurate, so could we scrape that at build (or on first view of the home page) instead using the official widget, in order to avoid hitting the API?

Reply:
I have tested your assumption. It holds. The stars counter decreases the counter. But I don't think that it will solve the shared IP issue of large enterprises. We are not the only website that might use the GitHub API.

Reply:
Cache the API result client side, and only hit the API every 24 hours. Fall back to the cache if the API request fails?

Reply:
What about this strategy? We create a new GitHub user, e.g. with this email `netlify@material-ui.com`. We leave this user with no access to org or private repos. We create a personal `public_repo` token.

Reply:
Oops, simpler:
![Capture d’écran 2020-01-17 à 16 34 47](https://user-images.githubusercontent.com/3165635/72624461-4d443b80-3947-11ea-95dc-53b38810822c.png)

I have updated the Netlify env accordingly, problem sovled :)

Reply:
Does it sound good? @hiteshkundal Do you want to complete the effort? :)

Reply:
I guess we all stopped reading after "no scope" 😄

Reply:
> Does it sound good? @hiteshkundal Do you want to complete the effort? :)

sure